src/components/conversationalsearchresult/conversationalsearchresult.js

```diff
@@ -42,45 +42,54 @@ export default class ConversationalSearchResult {
-    const fadeOutOverlay = answerContainer.querySelector('.fade-out-overlay');
+    const fadeOutOverlay = answerContainer && answerContainer.querySelector('.fade-out-overlay');
     const showMoreBtn = document.querySelector('.show-more-btn');
-    const buttonText = showMoreBtn.querySelector('.button-text');
-    const chevron = showMoreBtn.querySelector('.chevron');
+    const buttonText = showMoreBtn && showMoreBtn.querySelector('.button-text');
+    const chevron = showMoreBtn && showMoreBtn.querySelector('.chevron');
 
-    if (answerContainer && showMoreBtn) {
-      answerContainer.style.maxHeight = `${this.answerMaxHeight}px`;
+    if (!answerContainer || !fadeOutOverlay || !showMoreBtn || !buttonText || !chevron) {
+      return;
+    }
 
-      if (answerContainer.scrollHeight > this.answerMaxHeight) {
-        showMoreBtn.style.display = 'flex';
-        fadeOutOverlay.style.display = 'block';
-      } else {
-        showMoreBtn.style.display = 'none';
-        fadeOutOverlay.style.display = 'none';
-      }
+    answerContainer.style.maxHeight = `${this.answerMaxHeight}px`;
 
-      showMoreBtn.addEventListener('click', () => {
-        if (answerContainer.classList.contains('collapsed')) {
-          answerContainer.style.maxHeight = `${answerContainer.scrollHeight}px`;
-          answerContainer.classList.remove('collapsed');
-          buttonText.textContent = 'Show less';
-          fadeOutOverlay.style.display = 'none';
-          chevron.style.transform = 'rotate(180deg)';
-        } else {
-          answerContainer.style.maxHeight = `${this.answerMaxHeight}px`;
-          answerContainer.classList.add('collapsed');
-          buttonText.textContent = 'Show more';
-          fadeOutOverlay.style.display = 'block';
-          chevron.style.transform = '';
-        }
-      });
+    if (answerContainer.scrollHeight > this.answerMaxHeight) {
+      showMoreBtn.style.display = 'flex';
+      fadeOutOverlay.style.display = 'block';
+    } else {
+      showMoreBtn.style.display = 'none';
+      fadeOutOverlay.style.display = 'none';
     }
+
+    showMoreBtn.addEventListener('click', () => {
+      if (answerContainer.classList.contains('collapsed')) {
+        answerContainer.style.maxHeight = `${answerContainer.scrollHeight}px`;
+        answerContainer.classList.remove('collapsed');
+        buttonText.textContent = 'Show less';
+        fadeOutOverlay.style.display = 'none';
+        chevron.style.transform = 'rotate(180deg)';
+      } else {
+        answerContainer.style.maxHeight = `${this.answerMaxHeight}px`;
+        answerContainer.classList.add('collapsed');
+        buttonText.textContent = 'Show more';
+        fadeOutOverlay.style.display = 'block';
+        chevron.style.transform = '';
+      }
+    });
   }
 
   setupActionButtons() {
     const conversationalSearchResultContainer = document.querySelector(
       '.addsearch-conversational-search-result'
```

Comment:
Another way to select `.addsearch-conversational-search-result` without depending on the template is using `containerId` from component's setting.
```
const conversationalSearchResultContainer = document.querySelector(`#${this.conf.containerId}`)
```

Reply:
Good point. But I think we might as well keep some fixed css selector names. If anyone would want to style this using their own template they will have to work with certain fixed css selector names anyways (e.g. copy-btn, etc.). 

```const conversationalSearchResultContainer = document.querySelector(
      '.addsearch-conversational-search-result'
    );

    if (!conversationalSearchResultContainer) {
      return;
    }

    const copyButton =
      conversationalSearchResultContainer &&
      conversationalSearchResultContainer.querySelector('.copy-btn');
    const copyConfirm = conversationalSearchResultContainer.querySelector('.copy-confirm-message');
    const thumbsUpButton = conversationalSearchResultContainer.querySelector('.thumbs-up-btn');
    const thumbsDownButton = conversationalSearchResultContainer.querySelector('.thumbs-down-btn');```